src/Cli/dotnet/Commands/Run/VirtualProjectBuildingCommand.cs

```diff
@@ -1061,6 +1110,44 @@ private Package() { }
             };
         }
     }
+
+    /// <summary>
+    /// <c>#:project</c> directive.
+    /// </summary>
+    public sealed class Project : Named
+    {
+        private Project() { }
+
+        public static Project Parse(ImmutableArray<SimpleDiagnostic>.Builder? errors, SourceFile sourceFile, TextSpan span, string directiveText)
+        {
+            try
+            {
+                // If the path is a directory like '../lib', transform it to a project file path like '../lib/lib.csproj'.
+                // Also normalize blackslashes to forward slashes to ensure the directive works on all platforms.
+                var sourceDirectory = Path.GetDirectoryName(sourceFile.Path) ?? ".";
+                var resolvedProjectPath = Path.Combine(sourceDirectory, directiveText.Replace('\\', '/'));
+                if (Directory.Exists(resolvedProjectPath))
+                {
+                    var fullFilePath = MsbuildProject.GetProjectFileFromDirectory(resolvedProjectPath).FullName;
```

Comment:
A few lines above we normalize `\\` to `/` but `FullName` on windows will still have `\\` in the path. Should we re-normalize here?

Reply:
Consider user writing `#:project ..\lib` on Windows and another user executing that on Linux. The normalization above makes sure that's possible.

Normalization here isn't necessary for that to work, but I guess we could normalize for other reasons. But the current behavior (without normalization) should be consistent with what `dotnet reference add` is doing (since we use the same utility `MsbuildProject.GetProjectFileFromDirectory` as that command).

Reply:
`dotnet run --project ..\lib` probably wouldn't work. But `<ProjectReference Include="..\lib\lib.csproj" />` would.

Reply:
Yep MSBuild "solves" this for us as paths there are normalized. We can decide to simply pass through what the user types in the directive directly I think.

Reply:
> We can decide to simply pass through what the user types in the directive directly I think.

It's not so simple since we also want to support users specifying _directory_ paths which `<ProjectReference/>` elements don't support, therefore we might need to search for the project file in the directory and for that to work on linux, we need to normalize slashes.